Code/Framework/AzNetworking/AzNetworking/UdpTransport/UdpHeartbeatThread.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzNetworking/Utilities/TimedThread.h>
+
+namespace AzNetworking
+{
+    class UdpNetworkInterface;
+
+    //! @class UdpHeartbeatThread
+    //! @brief Sends a heartbeat packet if the main-thread system tick has been blocked and unable to process packets in a while.
+    class UdpHeartbeatThread
+        : public TimedThread
+    {
+    public:
+
+        UdpHeartbeatThread();
+        ~UdpHeartbeatThread() override;
+        void RegisterNetworkInterface(UdpNetworkInterface* udpNetworkInterface);
+        void UnregisterNetworkInterface(UdpNetworkInterface* udpNetworkInterface);
+
+    private:
+
+        //! Helper to determine if a given socket is monitored by this reader thread instance
+        void OnStart() override;
+        void OnStop() override;
+        void OnUpdate(AZ::TimeMs updateRateMs) override;
+
+        AZ_DISABLE_COPY_MOVE(UdpHeartbeatThread);
+
+        AZStd::vector<UdpNetworkInterface*> m_udpNetworkInterfaces;
```

Comment:
Do we need multiple network interfaces?  Is there a situation where we are connected to multiple servers?

Reply:
Yep! This can even be used for servers connected to other servers (multi-server setup).